### ISSUE TITLE
Automate Templates dependency flow

### ISSUE THREAD

**Issue by sebastienros** [dotnet/aspire#4853]
The project Templates use external dependencies like XUnit, MSTest, OpenTelemetry, ...
These dependencies have a mix of automated versions like `REPLACE_WITH_EXTENSIONS_VERSION` and some static values. The static values are prone to be stale since they are not updated by dependabot while we need to use recent versions at release.

Furthermore, we have a `cgmanifest.json` file which lists all these dependencies such that Component Governance can identify issues for the dlls (https://github.com/dotnet/aspire/pull/4841) that are not part of the generated artifacts. This has to be kept in sync with all the package references and their versions we use in Templates.

The suggestion is to drive all the versions for the templates and `cgmanifest.json` in `Directory.Packages.props` and keep it as the central version management location. We already have a Target in `AspireTemplates.json` to do some version replacement. This will also allow work with dependabot.

**Comment by davidfowl** [comment]
Is this still valid @DamianEdwards @sebastienros @joperezr ?

**Comment by joperezr** [comment]
I don't think we need cgmanifest any longer as we have been doing work to get the templates themselves building into our official builds and CG itself is now inspecting it. @radical could you confirm whether or not this is still needed and removing if not?

**Comment by radical** [comment]
Yes, we are building the templates, and exercising them via tests which allows CG tools to extract the dependencies. IIUC, and this is the real ask then this can be closed.
